### PR TITLE
[RFC] eval.c: Add ga_concat_esc function Issue#7256

### PR DESCRIPTION
**Pull request by acferreir4**
Add ga_concat_esc present in vim-patch: vim/vim@2368917

### CODE REVIEW

**Comment by brcolow** [comment]
You should mark 1325 as applied now in `version.c`.

File: src/nvim/eval.c [hunk 1]
```diff
@@ -6733,6 +6733,39 @@ static void prepare_assert_error(garray_T *gap)
...
+    char_u *p;
+    char_u buf[NUMBUFLEN];
+
+    if (str == NULL) {
+	ga_concat(gap, (char_u *)"NULL");
+	return;
+    }
+
+    for (p = str; *p != NUL; ++p)
+	switch (*p) {
```
Comment (justinmk) [inline src/nvim/eval.c]: need to fix the indent. Try `make lint` on your local machine to see any other lint issues.

**Comment by justinmk** [comment]
is this still WIP?

**Comment by acferreir4** [comment]
I'll sort out the linting now, thanks

**Comment by acferreir4** [comment]
Not entirely sure what broke the quickbuild in my last commit, my newb status is certainly showing.

**Comment by KillTheMule** [comment]
QB failure looks very much unrelated.

**Comment by acferreir4** [comment]
Seems related to a prompt expecting an 'enter' press. Is this something I can fix or should I go ahead and do the rebase/squash step and mark this RDY?

**Comment by KillTheMule** [comment]
RDY means the PR had a review, so I guess you should leave it at RFC.

**Pull request merged**